projects/cps-ui-kit/src/lib/components/cps-tree-table/directives/internal/tree-table-unsort.directive.ts

```diff
@@ -263,47 +280,44 @@ export class TreeTableUnsortDirective {
               !globalMatch &&
               globalFilterFieldsArray
             ) {
-              for (let j = 0; j < globalFilterFieldsArray.length; j++) {
-                const copyNodeForGlobal = { ...copyNode };
-                const filterField =
-                  globalFilterFieldsArray[j].field ||
-                  globalFilterFieldsArray[j];
-                const filterValue = pTreeTable.filters.global.value;
-                const filterConstraint =
-                  pTreeTable.filterService.filters[
-                    pTreeTable.filters.global
-                      .matchMode as keyof typeof pTreeTable.filterService.filters
-                  ];
-                paramsWithoutNode = {
-                  filterField,
-                  filterValue,
-                  filterConstraint,
-                  isStrictMode
-                };
+              const copyNodeForGlobal = { ...copyNode };
+              const filterField = undefined;
+              const filterValue = pTreeTable.filters.global.value;
+              const filterConstraint =
+                pTreeTable.filterService.filters[
+                  pTreeTable.filters.global
+                    .matchMode as keyof typeof pTreeTable.filterService.filters
+                ];
+              paramsWithoutNode = {
+                filterField,
+                filterValue,
+                filterConstraint,
+                isStrictMode,
+                globalFilterFieldsArray
+              };
 
-                if (
-                  (isStrictMode &&
-                    (pTreeTable.findFilteredNodes(
-                      copyNodeForGlobal,
-                      paramsWithoutNode
-                    ) ||
-                      pTreeTable.isFilterMatched(
-                        copyNodeForGlobal,
-                        paramsWithoutNode as any
-                      ))) ||
-                  (!isStrictMode &&
-                    (pTreeTable.isFilterMatched(
+              if (
+                (isStrictMode &&
```

Comment:
@TerranceKhumalo-absa maybe we could create some variables for a better understanding of what exactly we are checking here